docs/HeliumParamSpecs.md

```diff
@@ -0,0 +1,295 @@
+    * Valid input range and does not contain results
+         * Status : 200
+         * Content-Type: application/json
+         * Response Body: Empty list []
+    * Invalid input - empty, foo, values not in valid range
+         * Status : 400
+         * Content-Type: text/plain
+         * Response Body: Invalid year parameter
+
+
+* Name : rating
+* Type: double
+* Parameter Validations:
+    * Valid input range containing results - [0.0, 10.0]
+    * Round off more than single decimal to a single decimal. Example: round off 7.253896 to 7.2.
+         * Status : 200
+         * Content-Type: application/json
+         * Response Body: List of movies filtered by rating as [{}, {},..]
+    * Valid input range and does not contain results
+         * Status : 200
+         * Content-Type: application/json
+         * Response Body: Empty list []
+    * Invalid input - empty, foo, values not in valid range [<0.0 , >10.0]
+         * Status : 400
+         * Content-Type: text/plain
+         * Response Body: Invalid rating parameter
+
+
+* Name : actorid
+* Type: string
+* Parameter Validations:
+    * Valid input - Wellformed actorid rule - starts with 'nm' followed by 9 digit integer converted to string. Example 'nm9877392'
+         * Status : 200
+         * Content-Type: application/json
+         * Response Body: List of movies filtered by actorid as [{}, {},..]
+    * Valid input - actorid conforms wellformed String rules and does not contain results
+         * Status : 200
+         * Content-Type: application/json
+         * Response Body: Empty list []
+    * Invalid input - empty, foo, values not conforming to wellformed actorid rule.  Example: 'nm1234', 'ab1234'
+         * Status : 400
+         * Content-Type: text/plain
+         * Response Body: Invalid actorid parameter
+
+
+* Name : genre
+* Type: string
+* Parameter Validations:
+    * Valid input - genre length >= 3. Example 'War' , 'Horror'
+         * Status : 200
+         * Content-Type: application/json
+         * Response Body: List of movies filtered by actorid as [{}, {},..]
+    * Valid input - genre length > 3 characters and does not contain results, example = horro
+         * Status : 200
+         * Content-Type: application/json
+         * Response Body: Empty list []
+    * Invalid input - empty, genre length < 3 characters
+         * Status : 400
+         * Content-Type: text/plain
+         * Response Body: Invalid genre parameter
+
+
+* Name : pagesize
+* Type: integer
+* Parameter Validations:
+    * Valid input range containing results - [1, 1000]
+         * Status : 200
+         * Content-Type: application/json
+         * Response Body: List of movies filtered by pagesize as [{}, {},..]
+    * Valid input range not containing results - [1, 1000]
+         * Status : 200
+         * Content-Type: application/json
+         * Response Body: Empty list []
+    * Invalid input - empty should picks defaults [1, 1000]
+         * Status : 200
```

Comment:
Should this return content? If the pagesize/pagenumber parameter is specified, but no value is provided - this feels like we'd want to return invalid input 400.

Reply:
For consistency, search will just return 200 [] representing not found.